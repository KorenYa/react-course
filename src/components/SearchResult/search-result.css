.search-result__no-hits {
    font-size: 60px;
    padding: 120px 20px;
    text-align: center;
}