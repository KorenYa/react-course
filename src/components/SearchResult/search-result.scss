@import '../../styles/constants.scss';

.search-result {
    font-size: 0px;

    &__no-hits {
        font-size: $x-lg;
        padding: 120px 20px;
        text-align: center;
    }
}
