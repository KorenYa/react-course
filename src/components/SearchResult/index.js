import React from 'react';
import MovieCard from '../MovieCard';
import './search-result.scss';
const SearchResult = function(props) {
    const moviesList = props.moviesList;

    const searchHits = moviesList.total > 0 && (
        <div className='search-result'>
            <div className='row'>
                {moviesList.data.map(movie => {
                    return (
                        <div className='col-4' key={movie.id}>
                            <MovieCard
                                date={movie.release_date.split('-')[0]}
                                genre={movie.genres[0]}
                                title={movie.title}
                                poster={movie.poster_path}
                            />
                        </div>
                    );
                })}
            </div>
        </div>
    );

    const noHits = <div className='search-result__no-hits'>No films found</div>;

    return moviesList.total > 0 ? searchHits : noHits;
};
export default SearchResult;
