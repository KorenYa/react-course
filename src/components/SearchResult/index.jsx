import React from 'react';
import PropTypes from 'prop-types';
import MovieCard from '../MovieCard';
import './search-result.scss';

const SearchResult = (props) => {
  const { moviesList } = props;

  const searchHits = moviesList.total > 0 && (
    <div className="search-result">
      <div className="row">
        {moviesList.data.map(movie => (
          <div className="col-4" key={movie.id}>
            <MovieCard
              id={movie.id}
              date={movie.release_date.split('-')[0]}
              genre={movie.genres[0]}
              title={movie.title}
              poster={movie.poster_path}
            />
          </div>
        ))}
      </div>
    </div>
  );

  const noHits = <div className="search-result__no-hits">No films found</div>;

  return moviesList.total > 0 ? searchHits : noHits;
};
SearchResult.propTypes = {
  moviesList: PropTypes.shape({
    data: PropTypes.array,
    total: PropTypes.number,
  }),
};
export default SearchResult;
