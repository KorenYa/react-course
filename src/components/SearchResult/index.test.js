import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import SearchResult from '../SearchResult';

describe('SearchResult', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(
                <MemoryRouter>
                    <SearchResult
                        moviesList={{
                            data: [
                                {
                                    id: 962,
                                    title: 'The Gold Rush',
                                    tagline:
                                        "The World's Greatest Laughing Picture!",
                                    vote_average: 7.8,
                                    vote_count: 411,
                                    release_date: '1925-06-25',
                                    poster_path:
                                        'https://image.tmdb.org/t/p/w500/eQRFo1qwRREYwj47Yoe1PisgOle.jpg',
                                    overview:
                                        'A lone prospector ventures into Alaska looking for gold. He gets mixed up with some burly characters and falls in love with the beautiful Georgia. He tries to win her heart with his singular charm.',
                                    budget: 923000,
                                    revenue: 2500000,
                                    genres: ['Adventure', 'Comedy', 'Drama'],
                                    runtime: 95
                                },
                                {
                                    id: 19,
                                    title: 'Metropolis',
                                    tagline:
                                        'There can be no understanding between the hands and the brain unless the heart acts as mediator.',
                                    vote_average: 8.1,
                                    vote_count: 837,
                                    release_date: '1927-01-10',
                                    poster_path:
                                        'https://image.tmdb.org/t/p/w500/qriaeUUwdmlgethK3aSAx68mG05.jpg',
                                    overview:
                                        "In a futuristic city sharply divided between the working class and the city planners, the son of the city's mastermind falls in love with a working class prophet who predicts the coming of a savior to mediate their differences.",
                                    budget: 92620000,
                                    revenue: 650422,
                                    genres: ['Drama', 'Science Fiction'],
                                    runtime: 153
                                },
                                {
                                    id: 143,
                                    title: 'All Quiet on the Western Front',
                                    tagline:
                                        'They left for war as boys never to return as men.',
                                    vote_average: 7.6,
                                    vote_count: 216,
                                    release_date: '1930-04-29',
                                    poster_path:
                                        'https://image.tmdb.org/t/p/w500/9gn5ce3B2vOpk9pOGeUjmFecIo5.jpg',
                                    overview:
                                        'A young soldier faces profound disillusionment in the soul-destroying horror of World War I. Together with several other young German soldiers, he experiences the horrors of war, such evil of which he had not conceived of when signing up to fight. They eventually become sad, tormented, and confused of their purpose.',
                                    budget: 1250000,
                                    revenue: 0,
                                    genres: ['Drama', 'War'],
                                    runtime: 133
                                }
                            ],
                            total: 3
                        }}
                    />
                </MemoryRouter>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
