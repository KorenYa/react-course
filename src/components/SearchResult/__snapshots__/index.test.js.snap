// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult renders correctly 1`] = `
<div
  className="search-result"
>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <div
        className="movie-card"
      >
        <div
          className="text-center"
        >
          <img
            alt="The Gold Rush"
            className="img-responsive"
            src="https://image.tmdb.org/t/p/w500/eQRFo1qwRREYwj47Yoe1PisgOle.jpg"
          />
        </div>
        <h2
          className="movie-card__title"
        >
          The Gold Rush
          <span
            className="movie-card__date"
          >
            1925
          </span>
        </h2>
        <div
          className="movie-card__genre"
        >
          Adventure
        </div>
      </div>
    </div>
    <div
      className="col-4"
    >
      <div
        className="movie-card"
      >
        <div
          className="text-center"
        >
          <img
            alt="Metropolis"
            className="img-responsive"
            src="https://image.tmdb.org/t/p/w500/qriaeUUwdmlgethK3aSAx68mG05.jpg"
          />
        </div>
        <h2
          className="movie-card__title"
        >
          Metropolis
          <span
            className="movie-card__date"
          >
            1927
          </span>
        </h2>
        <div
          className="movie-card__genre"
        >
          Drama
        </div>
      </div>
    </div>
    <div
      className="col-4"
    >
      <div
        className="movie-card"
      >
        <div
          className="text-center"
        >
          <img
            alt="All Quiet on the Western Front"
            className="img-responsive"
            src="https://image.tmdb.org/t/p/w500/9gn5ce3B2vOpk9pOGeUjmFecIo5.jpg"
          />
        </div>
        <h2
          className="movie-card__title"
        >
          All Quiet on the Western Front
          <span
            className="movie-card__date"
          >
            1930
          </span>
        </h2>
        <div
          className="movie-card__genre"
        >
          Drama
        </div>
      </div>
    </div>
  </div>
</div>
`;
